backend/tests/app_fixtures.py

```diff
@@ -8,6 +10,12 @@
 WEBSOCKET_URI = "ws://localhost:8000"
 
 
+@contextmanager
+def hypothesis_safe_redis(redis):
+    yield redis
```

Comment:
Hmm, I don't understand this. Is this a generator?

Reply:
It's a context manager (thing to be used in a with statement)
Normally you'd write that as a class with __enter__ and __exit__ methods. The decorator instead takes a generator function, things before the yield correspond to enter, things after the exit

Reply:
https://docs.python.org/3/library/contextlib.html#contextlib.contextmanager

Python docs are always hit or miss but there is this one